packages/react-log-viewer/src/LogViewer/logger.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useState, memo } from 'react';
+import { VariableSizeList as List, areEqual } from 'react-window';
+import { LoggerProps } from './LogViewer';
+import LoggerRow from './loggerRow';
+import LoggerToolbar from './loggerToolbar';
+import { useLoggerContext } from './LoggerContext';
+import memoize from 'memoize-one';
+import { LOGGER_HEIGHT, DEFAULT_FOCUS } from './utils/constants';
+import { isArrayOfString, searchForKeyword, parseConsoleOutput } from './utils/utils';
+import styles from '@patternfly/react-styles/css/components/LogViewer/logger.styles';
+
+const createLoggerDataItem = memoize(
+  (
+    parsedData: string[] | null,
+    loggerRef: React.RefObject<any>,
+    rowInFocus: number,
+    highlightedRowIndexes: number[],
+    setHighlightedRowIndexes: (indexes: number[]) => void
+  ) => ({
+    parsedData,
+    loggerRef,
+    rowInFocus,
+    highlightedRowIndexes,
+    setHighlightedRowIndexes
+  })
+);
+
+const Logger: React.FunctionComponent<LoggerProps> = memo(
+  ({ data = [''], customToolbarActions, dataSourceTitles }: any) => {
+    const [parsedData, setParsedData] = useState<string[] | null>([]);
+    const [dataSourcesAmount, setDataSourcesAmount] = useState<number>(1);
+    const loggerContext = useLoggerContext();
+    const {
+      rowInFocus,
+      setRowInFocus,
+      searchedInput,
+      highlightedRowIndexes,
+      setHighlightedRowIndexes,
+      currentDataSource,
+      setSearchedWordIndexes
+    } = loggerContext;
+    const DEFAULT_SEARCH_INDEX = 0;
+    const loggerRef = React.createRef<any>();
+    const dataToRender = createLoggerDataItem(
```

Comment:
Can't you just use `React.useMemo` instead?

Reply:
I was following the documentation on memoizing items going into the individual row renders, but I haven't found a way to memoize an object containing fields(duh) rather than memoizing a whole function with useMemo. 

If you know how to cook something like that up I could happily switch it, just haven't found the answer.

Would using memoize-one's helper function be a nono?

Reply:
It's not a nono, it's just preferred to not use unnecessary 3rd party deps. Is this not the same?
```js
const dataToRender = React.useMemo(() => ({
    parsedData,
    loggerRef,
    rowInFocus,
    highlightedRowIndexes,
    setHighlightedRowIndexes
}), [
    parsedData,
    loggerRef,
    rowInFocus,
    highlightedRowIndexes,
    setHighlightedRowIndexes
]);